lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,202 @@
+     */
+    protected $position;
+
+    /**
+     * @var Pdf
+     */
+    protected $snappy;
+
+    /**
+     * @throws \RuntimeException
+     *
+     * @param string $filename
+     * @param bool   $showHeaders
+     * @param bool   $showBorders
+     * @param array  $snappyOptions
+     * @param string $wkhtmltopdfBinary Path of the wkhtmltopdf binary
+     */
+    public function __construct($filename, $showHeaders = true, $showBorders = true, $snappyOptions = null, $wkhtmltopdfBinary = null)
+    {
+        $this->filename          = $filename;
+        $this->showHeaders       = $showHeaders;
+        $this->showBorders       = $showBorders;
+        $this->snappyOptions     = $snappyOptions;
+        $this->wkhtmltopdfBinary = $wkhtmltopdfBinary;
+        $this->position          = 0;
+        $this->snappy            = new Pdf();
+
+        if (is_file($filename)) {
+            throw new \RuntimeException(sprintf('The file %s already exist', $filename));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function open()
+    {
+        $this->html = '<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><meta name=ProgId content=Pdf.Document><meta name=Generator content="https://github.com/sonata-project/exporter"></head><body><table';
+        if (true === $this->showBorders) {
+            $this->html .= ' border="1"';
+        }
+        $this->html .= ' >';
+    }
+
+    /**
+     * @param $data
+     *
+     * @return array mixed
+     */
+    protected function init($data)
+    {
+        if ($this->position > 0) {
+            return;
+        }
+
+        if (true === $this->showHeaders) {
+            $this->html .= '<tr>';
+            foreach ($data as $header => $value) {
+                $this->html .= sprintf('<th>%s</th>', $header);
+            }
+            $this->html .= '</tr>';
+            ++$this->position;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function write(array $data)
+    {
+        $this->init($data);
+
+        $this->html .= '<tr>';
+        foreach ($data as $value) {
+            $this->html .= sprintf('<td>%s</td>', $value);
+        }
+        $this->html .= '</tr>';
+
+        ++$this->position;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close()
+    {
+        $this->html .= '</table></body></html>';
+        $this->generate();
+    }
+
+    /**
+     * Generates a Pdf document.
+     */
+    protected function generate()
+    {
+        $binaryPath = $this->getWkhtmltopdfBinary();
+        $this->snappy->setBinary($binaryPath);
+        $this->setOptions();
+        $this->snappy->generateFromHtml($this->html, $this->filename);
+    }
+
+    /**
+     * Retrieves wkhtmltopdf binary if it is not set.
+     *
+     * @return string $wkhtmltopdfBinary The path of wkhtmltopdf binary
+     */
+    protected function getWkhtmltopdfBinary()
+    {
+        if (is_file($this->wkhtmltopdfBinary)) {
```

Comment:
This should not be done here. However you can have a static function to try to find the binary. So the binary path is mandatory.